pallets/staking/src/pallet/mod.rs

```diff
@@ -1428,9 +1428,12 @@ pub mod pallet {
             if <Ledger<T>>::contains_key(&controller) {
                 return Err(Error::<T>::AlreadyPaired.into());
             }
-            // Prevents stashes which are controllers from calling the extrinsic
-            if <Ledger<T>>::contains_key(&stash) {
-                return Err(Error::<T>::BadState.into());
+
+            // Prevents stashes which are controllers of another ledger from calling the extrinsic
+            if let Some(bonded_ledger) = <Ledger<T>>::get(&stash) {
+                if bonded_ledger.stash != stash {
+                    return Err(Error::<T>::BadState.into());
+                }
```

Comment:
This will work.  But I think we can make it a little bit cheaper.  If `stash` and `old_controller` equal, then we can skip the ledger lookup.
```suggestion
            if old_controller != stash {
                if <Ledger<T>>::contains_key(&stash) {
                    return Err(Error::<T>::BadState.into());
                }
```